Code/C++/Doubly_linkedlist/Doubly_linkedlist_all_operations.cpp

```diff
@@ -0,0 +1,390 @@
+    node *ptr_1 = start->next;
+    int index;
+    cout << "\nEnter the index where you want to delete the node ";
+    cin >> index;
+    int count = 0;
+    if (index == 0)
+    {
+        deletion_at_beginning();
+    }
+    else if (index < 0)
+    {
+        cout << "\nInvalid index ";
+    }
+    else if (index == 4)
+    {
+
+        deletion_at_end();
+    }
+    else
+    {
+        while (count != index - 1)
+        {
+            ptr = ptr->next;
+            ptr_1 = ptr_1->next;
+            count += 1;
+        }
+        ptr->next = ptr_1->next;
+        ptr_1->next->prev = ptr;
+        free(ptr_1);
+    }
+}
+
+//Menu driven program to perform all the operations of doubly linked list.
+int main()
+{
+    Doubly_linklist D1;
+    int ch;
+read:
+    cout << "\n************Menu*************";
+    cout << "\n 1. create ";
+    cout << "\n 2. Insertion at beginning ";
+    cout << "\n 3. Insertion at end ";
+    cout << "\n 4. Insert at index ";
+    cout << "\n 5. Insert after index";
+    cout << "\n 6. Delete from beginning ";
+    cout << "\n 7. Delete from end ";
+    cout << "\n 8. Delete from specified location";
+    cout << "\n 9. Traverse ";
+    cout << "\nEnter your choice ";
+    cin >> ch;
+    switch (ch)
+    {
+    case 1:
+    {
+        D1.creation();
+        cout << "\n Node created successfully";
+        break;
+    }
+    case 2:
+    {
+        D1.insertion_at_beginning();
+        cout << "\n Node Inserted successfully at beginning";
+        break;
+    }
+    case 3:
+    {
+        D1.insertion_at_end();
+        cout << "\n Node Inserted successfully at End";
+        break;
+    }
+    case 4:
+    {
+        D1.insertion_at_index();
+        cout << "\n Node Inserted successfully at Given position";
+        break;
+    }
+    case 5:
+    {
+        D1.insertion_after_given_index();
+        cout << "\n Node Inserted successfully after a given index";
+        break;
+    }
+    case 6:
+    {
+        D1.deletion_at_beginning();
+        cout << "\n Node Deleted successfully from beginning";
+        break;
+    }
+    case 7:
+    {
+        D1.deletion_at_end();
+        cout << "\n Node Deleted successfully from End";
+        break;
+    }
+    case 8:
+    {
+        D1.deletion_at_index();
+        cout << "\n Node Deleted successfully from given position";
+        break;
+    }
+    case 9:
+    {
+        D1.traversal();
+        break;
+    }
+    }
+    int s;
+    cout << endl
+         << "\nDo you want to continue?" << endl;
+    cout << "\nPress 1 to continue and any other number to exit";
+    cin >> s;
+    if (s == 1)
+        goto read;
+    else
+        exit(0);
+    return 0;
+
+    /*complexity of traversing is
+   Time complexity:- O(n)
+   Space complexity:-O(n)
+   
+   complexity of inserting node 
+   Time complexity:- O(1) 
+   Space complexity:- O(1)
+   
+   complexity of deleting node 
+   Time complexity:- O(1) 
+   Space complexity:- O(1)
+   */
+}
```

Comment:
Add at least 2 input and output examples of this code before complexities (outside of the main())